### PR TITLE
Add support for range selection when holding shift key

### PR DESCRIPTION
**Pull request by tylerdiaz**
Implements https://github.com/fabiospampinato/notable/issues/44

### CODE REVIEW

**Comment by tylerdiaz** [comment]
@fabiospampinato Happy to help. I've updated the logic to support both cmd/ctrl selection and shift range selection.

File: src/renderer/containers/main/multi_editor.ts [hunk 1]
```diff
@@ -48,8 +48,36 @@ class MultiEditor extends Container<MultiEditorState, MainCTX> {
 
   }
 
+  _deduplicate ( arr: any[] = [] ) {
```
Comment (fabiospampinato) [inline src/renderer/containers/main/multi_editor.ts]: You should use `_.uniq` instead, but ideally this step shouldn't be needed.

File: src/renderer/containers/main/multi_editor.ts [hunk 2]
```diff
@@ -48,8 +48,36 @@ class MultiEditor extends Container<MultiEditorState, MainCTX> {
...
+
+      startIndex = markerNoteIndex, endIndex = anchorNoteIndex;
+
+    } else {
+
+      startIndex = anchorNoteIndex, endIndex = markerNoteIndex;
+
+    }
+
+    let notesNext = [...selectedNotes, ...notes.slice(startIndex, endIndex + 1)];
```
Comment (fabiospampinato) [inline src/renderer/containers/main/multi_editor.ts]: It looks like the previously selected notes are always preserved, this is not the behavior you can see [here](http://demo.getsvelto.com/widgets/selectable.html), notes in the range are instead toggled.

**Comment by tylerdiaz** [comment]
@fabiospampinato Good catch, I've addressed the logic to fit the example. The set theory approach is terse but perhaps harder to reason about than something like: 

```javascript
_.reject(selectedNotesRange, (note, index) => { 
  return _.find(selectedNotes, selectedNote => ({ selectedNote.id == note.id})) 
});
```

Any preference one way or the other?

**Comment by fabiospampinato** [comment]
@tylerdiaz as long as it works the way I want it to and the implementation isn't crazy it's ok for me. I probably like the sets approach better, unless it's significantly slower.

**Comment by fabiospampinato** [comment]
Thank you @tylerdiaz !

**Comment by fabiospampinato** [comment]
It does't work exactly like [this](http://demo.getsvelto.com/widgets/selectable.html) though, I'm gonna have to change it a little bit.

**Pull request merged**